lib/Models/CatalogItem.js

```diff
@@ -820,4 +853,33 @@ function isShownChanged(catalogItem) {
     catalogItem.terria.currentViewer.notifyRepaintRequired();
 }
 
+/**
+ * Toggle enable and disable attribution text
+ */
+
+function toggleAttributionText(catalogItem){
+    var terria = catalogItem.terria;
+    function add(textAttribution){
+        if (defined(terria.cesium)){
```

Comment:
No need for the Cesium/Leaflet check here.  Just do `terria.currentViewer.addAttributionText(textAttribution)`.  With that, I think you can eliminate this entire `toggleAttributionText` function and just call the above (and the remove version) directly.